OpenRA.Mods.Common/Traits/RejectsOrders.cs

```diff
@@ -44,10 +44,15 @@ public static bool AcceptsOrder(this Actor self, string orderString)
 			if (rejectsOrdersTraits.Length == 0)
 				return true;
 
-			var reject = rejectsOrdersTraits.SelectMany(t => t.Reject);
-			var except = rejectsOrdersTraits.SelectMany(t => t.Except);
+			foreach (var rejectsOrdersTrait in rejectsOrdersTraits)
+				if (rejectsOrdersTrait.Except.Contains(orderString))
+					return true;
 
-			return except.Contains(orderString) || (reject.Any() && !reject.Contains(orderString));
+			foreach (var rejectsOrdersTrait in rejectsOrdersTraits)
+				if (rejectsOrdersTrait.Reject.Contains(orderString))
+					return false;
+
+			return true;
```

Comment:
This needs to return `false` if there aren't any explicit rejects, doesn't it?

Reply:
Indeed, fixed.